quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/scheduler/SchedulerUtils.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.quarkus.scheduler;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for quarkus scheduler
+ *
+ */
+public class SchedulerUtils {
+
+	// Patterns for each cron part of the cron member expression in order of
+	// SchedulerErrorCode enum ordinal
+	// Note: DAY_OF_MONTH and DAY_OF_WEEK are mutually exclusive
+
+	private static final String SECOND_MINUTE_RANGE = "([1-5]?[0-9])";
+
+	private static final String HOUR_RANGE = "(2[0-3]|1?[0-9])";
+
+	private static final String DAY_OF_MONTH_RANGE = "(3[01]|[12][0-9]|[1-9])";
+
+	private static final String MONTH_RANGE = "(JAN|FEB|MA[RY]|APR|JU[NL]|AUG|SEP|OCT|NOV|DEC|1[0-2]|[1-9])";
+
+	private static final String DAY_OF_WEEK_RANGE = "(MON|TUE|WED|THU|FRI|SAT|SUN|[1-7])";
+
+	private static final String YEAR_RANGE = "(19[7-9][0-9]|20[0-9][0-9])";
+
+	private static final String CRON_SPECIAL_CHARACTERS = "|\\*|\\?|^\\*\\/\\d+";
+
+	private static final String BASE_CRON_PATTERN = "(%1$s(?:,%1$s)*|%1$s\\-%1$s|%2$s)";
```

Comment:
This accepts comma separated values and ranges if within the supported range. i.e. `3,4,5 * * * * *` and `3-5 * * * * *`  is accepted, but `3,4,60 * * * * *` and `3-60 * * * * *` is rejected.